src/main/java/org/nebula/contrib/ngbatis/session/IntervalCheckSessionDispatcher.java

```diff
@@ -36,15 +37,18 @@ public class IntervalCheckSessionDispatcher implements Runnable, SessionDispatch
    * 具备间隔时间做连接可用性检查的会话调度器
    * @param nebulaPoolConfig 连接信息
    */
-  public IntervalCheckSessionDispatcher(NebulaPoolConfig nebulaPoolConfig) {
+  public IntervalCheckSessionDispatcher(NebulaPoolConfig nebulaPoolConfig, ParseCfgProps cfgProps) {
     this.nebulaPoolConfig = nebulaPoolConfig;
     this.sessionQueue = new ArrayBlockingQueue<>(nebulaPoolConfig.getMaxConnSize());
     threadPool = EnvConfig.reconnect ? Executors.newScheduledThreadPool(1) : null;
+    //使用自定义的 session存活有效期/健康检测间隔
+    SESSION_LIFE_LENGTH = cfgProps.getSessionLifeLength() == null ? SESSION_LIFE_LENGTH : cfgProps.getSessionLifeLength();
+    CHECK_FIXED_RATE = cfgProps.getCheckFixedRate() == null ? CHECK_FIXED_RATE : cfgProps.getCheckFixedRate();
     wakeUp();
   }
 
   @Override
-  public void run() {
+  public synchronized void run() {

```

Comment:
In constructor, the thread pool only keeps 1 thread, and only this thread invokes `run` method.
Maybe it's not necessary to be synchronized.

Reply:
Use synchronized to lock the shared variable sessionQueue of the run method and poll method

Reply:
Does the problem recur in your use case?

Reply:
In the current unit test, there are no corresponding exceptions